R-proj/src/sample_points.cpp

```diff
@@ -0,0 +1,322 @@
+                    Sam_arb_simplex(VP, numpoints, randPoints);
+                } else {
+                    throw Rcpp::exception("Not a simplex!");
+                }
+            } else if (Rcpp::as<bool>(exact) && type!=2) {
+                throw Rcpp::exception("Not a simplex in V-representation!");
+            }
+        } else {
+            if (!body.isNotNull()) {
+
+                throw Rcpp::exception("Wrong input!");
+
+            } else if (!Rcpp::as<Rcpp::List>(Parameters).containsElementNamed("dimension")) {
+
+                throw Rcpp::exception("Wrong input!");
+
+            }
+            dim = Rcpp::as<NT>(Rcpp::as<Rcpp::List>(Parameters)["dimension"]);
+            if (Rcpp::as<Rcpp::List>(Parameters).containsElementNamed("radius")) {
+
+                radius = Rcpp::as<NT>(Rcpp::as<Rcpp::List>(Parameters)["radius"]);
+
+            }
+            if (Rcpp::as<std::string>(body).compare(std::string("hypersphere"))==0) {
+
+                for (unsigned int k = 0; k < numpoints; ++k) {
+                    randPoints.push_back(get_point_on_Dsphere<RNGType , Point >(dim, radius));
+                }
+
+            } else if (Rcpp::as<std::string>(body).compare(std::string("ball"))==0) {
+
+                for (unsigned int k = 0; k < numpoints; ++k) {
+                    randPoints.push_back(get_point_in_Dsphere<RNGType , Point >(dim, radius));
+                }
+
+            } else if (Rcpp::as<std::string>(body).compare(std::string("unit simplex"))==0) {
+
+                Sam_Unit<NT, RNGType >(dim, numpoints, randPoints);
+
+            } else if (Rcpp::as<std::string>(body).compare(std::string("canonical simplex"))==0) {
+
+                Sam_Canon_Unit<NT, RNGType >(dim, numpoints, randPoints);
+
+            } else {
+
+                throw Rcpp::exception("Wrong input!");
+
+            }
+        }
+    } else if (P.isNotNull()) {
+
+        type = Rcpp::as<Rcpp::Reference>(P).field("type");
+        dim = Rcpp::as<Rcpp::Reference>(P).field("dimension");
+        unsigned int walkL = 10+dim/10;
+        Point MeanPoint;
+        if (InnerPoint.isNotNull()) {
+            if (Rcpp::as<Rcpp::NumericVector>(InnerPoint).size()!=dim) {
+                Rcpp::warning("Internal Point has to lie in the same dimension as the polytope P");
+            } else {
+                set_mean_point = true;
+                MeanPoint = Point( dim , Rcpp::as<std::vector<NT> >(InnerPoint).begin(),
+                        Rcpp::as<std::vector<NT> >(InnerPoint).end() );
+            }
+        }
+        if(walk_length.isNotNull()) walkL = Rcpp::as<unsigned int>(walk_length);
+
+        NT a = 0.5;
+
+        if (Rcpp::as<Rcpp::List>(Parameters).containsElementNamed("variance"))
+            a = 1.0 / (2.0 * Rcpp::as<NT>(Rcpp::as<Rcpp::List>(Parameters)["variance"]));
+
+        if(!WalkType.isNotNull() || Rcpp::as<std::string>(WalkType).compare(std::string("CDHR"))==0){
```

Comment:
see comment above. Also consider that comment in all the cases that you use that kind of checks

Reply:
done.